### PR TITLE
WIP Chess, Recorder, cell fill detection

### PR DESCRIPTION
**Pull request by danirukun**
- Classes for board and pieces
- Initial board state
- WIP filled cell detection from board image matrix canny window (accuracy > 99%)
- Basic PGN tag writer from interactive and system data input

### CODE REVIEW

File: src/Recorder.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Recorder::Recorder()
+{
+    outfile.open("game.pgn");
+    aic::Log(aic::DBG) << "Opened file for writing.";
+}
+
+void Recorder::writeTagsInteractive()
+{
+    char data[100];
+    std::cout << "Please enter event name: " << std::endl;
```
Comment (agriic) [inline src/Recorder.cpp]: you can not use std::cout directly - it is data race.

File: src/main.cpp [hunk 2]
```diff
@@ -10,7 +10,7 @@ int main(int argc, const char* argv[])
 {
     Log::threshold = ALL;
 
-    std::string path = "0"; // WebCam
+    std::string path = "/Users/danpetrov/GIT/chess-vr/samples/chess.avi"; // WebCam
```
Comment (agriic) [inline src/main.cpp]: Use Xcode comandline arguments or don't add absolute paths.

**Review by agriic — CHANGES_REQUESTED** [review]

File: src/Recorder.cpp (hunk 1 above)
Comment (danirukun) [reply src/Recorder.cpp]: Make some other wrapper or use some configuration file? Command line arguments could be a good replacement.

Comment (agriic) [reply src/Recorder.cpp]: In general i think config file will be needed, but for now, can replace std::cout with Log(DBG).
And please place all classes under namespace "aic"

File: src/main.cpp (hunk 2 above)
Comment (danirukun) [reply src/main.cpp]: Removed abs path.

File: src/Recorder.cpp (hunk 1 above)
Comment (danirukun) [reply src/Recorder.cpp]: Changed to Log and put new classes under `aic`

File: src/Chess.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+aic::Piece::Piece(Type t, Piece::Color col, char colum, int lvl)
+{
+    this->type = t;
+    this->color = col;
+    this->column = colum;
+    this->level = lvl;
+}
+
+void aic::Piece::setPieceType(Piece::Type t)
```
Comment (agriic) [inline src/Chess.cpp]: Type and color will never change - no need for setters.

File: src/Chess.cpp [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+{
+    type = t;
+}
+
+void aic::Piece::setPieceColor(Piece::Color col)
+{
+    color = col;
+}
+
+void aic::Piece::setPieceColumn(char colum)
```
Comment (agriic) [inline src/Chess.cpp]: Chage piece position in one call, no two seperate

File: src/Chess.cpp [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+#include <utility>
+
+namespace aic
+{
+
+aic::Board::Board() = default;
+
+aic::Piece::Piece(Type t, Piece::Color col, char colum, int lvl)
+{
+    this->type = t;
```
Comment (agriic) [inline src/Chess.cpp]: No need for "this->"

File: src/Chess.cpp [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+void aic::Board::addPiece(Piece piece)
+{
+    boardState.push_back(piece);
+}
+
+void aic::Board::addPieces(const std::vector<Piece>& pieces)
+{
+    for (auto piece : pieces) {
```
Comment (agriic) [inline src/Chess.cpp]: use `boardState.insert(boardState.end(), pieces.begin(), pieces.end());`

File: src/Chess.hpp [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+     * Set the piece level location.
+     *
+     * @param lvl Level (row) as int (1-8).
+     * @return void.
+     */
+    void setPieceLevel(int lvl);
+
+    Piece::Type getPieceType()
+    {
+        return this->type;
```
Comment (agriic) [inline src/Chess.hpp]: don't uses `this` where it is really needed

File: src/Chess.hpp [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+    int getPieceLvl()
+    {
+        return this->level;
+    }
+
+private:
+    Piece::Type type;
+    Piece::Color color;
+    char column;
```
Comment (agriic) [inline src/Chess.hpp]: Lets switch to chess official terminology - columns are files and rows are ranks

File: src/Chess.cpp [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    addPiece(Piece(Piece::Type::KNIGHT, Piece::Color::WHITE, 'G', 1));
+
+    addPiece(Piece(Piece::Type::BISHOP, Piece::Color::WHITE, 'C', 1));
+    addPiece(Piece(Piece::Type::BISHOP, Piece::Color::WHITE, 'F', 1));
+
+    addPiece(Piece(Piece::Type::QUEEN, Piece::Color::WHITE, 'D', 1));
+    addPiece(Piece(Piece::Type::KING, Piece::Color::WHITE, 'E', 1));
+
+    /* Black pieces */
+    addPiece(Piece(Piece::Type::ROOK, Piece::Color::BLACK, 'A', 1));
```
Comment (agriic) [inline src/Chess.cpp]: Wrong positions for black pieces

**Review by agriic — CHANGES_REQUESTED** [review]

File: src/Chess.hpp (hunk 8 above)
Comment (danirukun) [reply src/Chess.hpp]: Renamed.

File: src/Chess.cpp (hunk 9 above)
Comment (danirukun) [reply src/Chess.cpp]: Fixed.

File: src/Chess.cpp (hunk 4 above)
Comment (danirukun) [reply src/Chess.cpp]: Reimplemented as `setPiecePos()`

File: src/Chess.cpp (hunk 6 above)
Comment (danirukun) [reply src/Chess.cpp]: Replaced.

File: src/Chess.cpp (hunk 3 above)
Comment (danirukun) [reply src/Chess.cpp]: Removed redundant methods.

**Review by agriic — APPROVED** [review]

**Pull request merged**